test/aleph/http_test.clj

```diff
@@ -238,6 +245,28 @@
             @(http-get (str "https://localhost:" port "/" path)))))
        (str path "path failed")))))
 
+(deftest test-ssl-files
+  (let [client-url (str "https://localhost:" port)
+        client-options {:connection-options {:ssl-context ssl/client-ssl-context}}
+        client-pool    (http/connection-pool client-options)]
+   (with-ssl-handler identity (merge default-ssl-options {:ssl-context ssl/server-ssl-context})
+     (is (empty?
+          (bs/to-string
+           (:body @(http-put client-url
+                             {:body (io/file "test/empty.txt")
+                              :pool client-pool})))))
+     (is (= (slurp "test/file.txt" :encoding "UTF-8")
+            (bs/to-string
+             (:body @(http-put client-url
+                               {:body (io/file "test/file.txt")
+                                :pool client-pool})))))
+     (is (thrown-with-msg? Exception
+                             #"BOOM!"
+                             (with-redefs [core/send-file-body (fn [_ _ _ _] (throw (Exception. "BOOM!")))]
+                                    @(http-put client-url
+                                     {:body (io/file "test/file.txt")
+                                      :pool client-pool})))))))
```

Comment:
This test is too fragile for the value it provides, I think. I'd rather not incorporate it.